extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchAggregatorFactory.java

```diff
@@ -0,0 +1,304 @@
+    Util.checkIfPowerOf2(this.nominalEntries, "nominalEntries");
+    this.metricColumns = metricColumns;
+    this.numberOfValues = numberOfValues == null ? (metricColumns == null ? 1 : metricColumns.size()) : numberOfValues;
+    if (metricColumns != null && metricColumns.size() != this.numberOfValues) {
+      throw new IAE("Number of metricColumns (" + metricColumns.size() + ") must agree with numValues ("
+          + this.numberOfValues + ")");
+    }
+  }
+
+  @Override
+  public Aggregator factorize(final ColumnSelectorFactory metricFactory)
+  {
+    if (metricColumns == null) {
+      final BaseObjectColumnValueSelector<ArrayOfDoublesSketch> selector = metricFactory
+          .makeColumnValueSelector(fieldName);
+      if (selector instanceof NilColumnValueSelector) {
+        return new ArrayOfDoublesSketchNoOpAggregator(numberOfValues);
+      }
+      return new ArrayOfDoublesSketchMergeAggregator(selector, nominalEntries, numberOfValues);
+    }
+    final DimensionSelector keySelector = metricFactory
+        .makeDimensionSelector(new DefaultDimensionSpec(fieldName, fieldName));
+    if (keySelector instanceof NullDimensionSelector) {
+      return new ArrayOfDoublesSketchNoOpAggregator(numberOfValues);
+    }
+    final List<BaseDoubleColumnValueSelector> valueSelectors = new ArrayList<>();
+    for (final String column: metricColumns) {
+      final BaseDoubleColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(column);
+      valueSelectors.add(valueSelector);
+    }
+    return new ArrayOfDoublesSketchBuildAggregator(keySelector, valueSelectors, nominalEntries);
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(final ColumnSelectorFactory metricFactory)
+  {
+    if (metricColumns == null) {
+      final BaseObjectColumnValueSelector<ArrayOfDoublesSketch> selector = metricFactory
+          .makeColumnValueSelector(fieldName);
+      if (selector instanceof NilColumnValueSelector) {
+        return new ArrayOfDoublesSketchNoOpBufferAggregator(numberOfValues);
+      }
+      return new ArrayOfDoublesSketchMergeBufferAggregator(selector, nominalEntries, numberOfValues,
+          getMaxIntermediateSize());
+    }
+    final DimensionSelector keySelector = metricFactory
+        .makeDimensionSelector(new DefaultDimensionSpec(fieldName, fieldName));
+    if (keySelector instanceof NullDimensionSelector) {
+      return new ArrayOfDoublesSketchNoOpBufferAggregator(numberOfValues);
+    }
+    final List<BaseDoubleColumnValueSelector> valueSelectors = new ArrayList<>();
+    for (final String column: metricColumns) {
+      final BaseDoubleColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(column);
+      valueSelectors.add(valueSelector);
+    }
+    return new ArrayOfDoublesSketchBuildBufferAggregator(keySelector, valueSelectors, nominalEntries,
+        getMaxIntermediateSize());
```

Comment:
This is not according to Druid style. In Druid style, either all method/constructor call args fit one line: `foo(a, b, c);`, or, if it doesn't fit 120 cols, it is formatted like this:
```java
foo(
    a,
    b,
    c
);
```
i. e. each arg on it's own line, and the closing paren also on it's own line.

There are many violations of this in this PR, please fix them.